client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -217,14 +212,16 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             final ProtonConnection con,
             final String sourceAddress,
             final ProtonQoS qos,
+            final int prefetchCredits,
             final ProtonMessageHandler messageHandler,
             final Handler<String> closeHook) {
 
         Future<ProtonReceiver> result = Future.future();
         ctx.runOnContext(go -> {
             final ProtonReceiver receiver = con.createReceiver(sourceAddress);
             receiver.setAutoAccept(true);
-            receiver.setPrefetch(DEFAULT_SENDER_CREDITS);
+            receiver.setQoS(qos);
+            receiver.setPrefetch(prefetchCredits);
```

Comment:
sanity checks? prefetchCredits > 0

Reply:
Vertx proton checks for >= 0. 0 might be useful for own flow control?

Reply:
yes, still: what happens if < 0? if an exception is thrown, it should be declared in the JavaDocs